src/transfer.jl

```diff
@@ -153,3 +121,49 @@ function transfer_blochwave(ψ_in, basis_in::PlaneWaveBasis{T},
         transfer_blochwave_kpt(ψ_in[ik], basis_in, basis_in.kpoints[ik], basis_out, kpt_out)
     end
 end
+
+"""
+Find the equivalent index of the coordinate `kcoord` ∈ ℝ³ in a list `kcoords` ∈ [-½, ½)³.
+`ΔG` is the vector of ℤ³ such that `kcoords[index] = kcoord + ΔG`.
+"""
+function find_equivalent_kpt(basis::PlaneWaveBasis{T}, kcoord, spin; tol=sqrt(eps(T))) where {T}
+    kcoord_red = map(kcoord) do k
+                     k = mod(k, 1)              # coordinate in [0, 1)³
+                     k ≥ 0.5 - tol ? k - 1 : k  # coordinate in [-½, ½)³
+                 end
+
+    ΔG = kcoord_red - kcoord
+    # ΔG should be an integer.
+    @assert all(is_approx_integer.(ΔG))
+    ΔG = round.(Int, ΔG)
+
+    indices_σ = krange_spin(basis, spin)
+    kcoords_σ = getfield.(basis.kpoints[indices_σ], :coordinate)
+    # Unique by construction.
+    index::Int = findfirst(isapprox(kcoord_red; atol=tol), kcoords_σ) + (indices_σ[1] - 1)
+
+    return (; index, ΔG)
+end
+
+"""
+Return the Fourier coefficients for `ψk · e^{i q·r}` in the basis of `kpt_out`, where `ψk`
+is defined on a basis `kpt_in`.
+"""
+function multiply_by_expiqr(basis, kpt_in, q, ψk)
+    shifted_kcoord = kpt_in.coordinate .+ q  # coordinate of ``k``-point in ℝ
+    index, ΔG = find_equivalent_kpt(basis, shifted_kcoord, kpt_in.spin)
+    kpt_out = basis.kpoints[index]
+    return transfer_blochwave_kpt(ψk, basis, kpt_in, kpt_out, ΔG)
+end
+
+"""
+Return an `ordering` function that reorders the `kpoints`. That is for each `kpoint` of the
+`basis`: `kpoints[ik].coordinate + q = ordering(kpoints)[ik].coordinate`.
+"""
+function kpoints_ordering(basis::PlaneWaveBasis, qcoord)
```

Comment:
I would call it k_to_kpq_mapping maybe?? We used "mapping" for this kind of transfer array in other parts of the code.

Reply:
Otherwise yes I like the array much better, thanks!